modules/compression/include/ips_jpg_at_model.cpp

```diff
@@ -0,0 +1,226 @@
+		i_row[col_signal.read()] = double(PixelValue_signal.read());
+	  }
+  }
+
+  //void OutputPixel(int *Pixel, int row, int col) {
+  //  double* i_row = &image[row * image_cols];
+  //  *Pixel = int(i_row[col]);
+  //}
+
+  void OutputByte() {
+	output_event.notify(8, SC_NS);
+  }
+
+  void output_operation(){
+	  while(true) {
+		wait(output_event);
+        Element_signal = image[index_signal.read()];
+	  }
+  }
+
+  void JPEG_compression() {
+	  compression_event.notify(100, SC_NS);
+  }
+
+  void compression_operation() {
+    while(true) {
+		wait(compression_event);
+        int output_size = 0;
+		//Level shift
+		for(int i=0; i<(image_rows*image_cols);i++){
+			image[i]=image[i]-128;
+		}
+		wait(100, SC_NS);
+		int Number_of_blocks = image_rows*image_cols/(Block_rows*Block_cols);
+		int block_output[Number_of_blocks][Block_rows*Block_cols] = {0};
+		int block_output_size[Number_of_blocks] = {0};
+		int block_counter = 0;
+		output_size = 0;
+		for(int row=0; row<image_rows; row+=Block_rows)	{
+		  double* i_row = &image[row * image_cols];
+		  for(int col=0; col<image_cols; col+=Block_cols) { //Divided the image in 8×8 blocks
+			DCT(row,col);
+			Quantization(row,col);
+			ZigZag(row,col,&block_output_size[block_counter],block_output[block_counter]);
+			output_size += block_output_size[block_counter]+1;
+			block_counter++;
+		  }
+		}
+		int output_counter = 0;
+		for(int block_index=0;block_index<Number_of_blocks;block_index++){
+		  for(int out_index=0; out_index<block_output_size[block_index];out_index++){
+			image[output_counter]=block_output[block_index][out_index];
+			output_counter++;
+		  }
+		  image[output_counter]=EOB;
+		  output_counter++;
+		}
+		output_size_signal = output_size;
+	}
+  }  
+  
+  void DCT(int row_offset, int col_offset) {
+    wait(400, SC_NS);
+	double cos_table[Block_rows][Block_cols];
+	for (int row = 0; row < Block_rows; row++) //make the cosine table
+	{
+      for (int col = 0; col < Block_cols; col++) {
+        cos_table[row][col] = cos((((2*row)+1)*col*PI)/16);
+	  }
+	}
+    double temp;
+	for(int row=row_offset; row<row_offset+Block_rows; row++)
+	{
+ 	  double* i_row = &image[row * image_cols];
+	  for(int col=col_offset; col<col_offset+Block_cols; col++) {
+		//i_row[col] = cos_table[row-row_offset][col-col_offset];
+		temp = 0.0;
+    	for (int x = 0; x < 8; x++){
+    		double* x_row = &image[(x+row_offset) * image_cols];
+    		for (int y = 0; y < 8; y++) {
+    		  temp += x_row[y+col_offset] * cos_table[x][row-row_offset] * cos_table[y][col-col_offset];
+    		}
+    	}
+    	if ((row-row_offset == 0) && (col-col_offset == 0)) {
+		  temp /= 8.0;
+		}
+    	else if (((row-row_offset == 0) && (col-col_offset != 0)) || ((row-row_offset != 0) && (col-col_offset == 0))){
+    	  temp /= (4.0*sqrt(2.0)); 
+    	}
+        else {
+    	  temp /= 4.0;
+        }
+		i_row[col] = temp;
+      }
+	}
+  }
+
+  void Quantization(int row_offset, int col_offset) {
```

Comment:
Use consistent notation for methods and variable names like camelCase, snake_case, or PascalCase.